### PR TITLE
Adding 1.2.5 manifest

### PR DESCRIPTION
**Pull request by abhinavGupta16**
### Description
Adding manifest files for `1.2.5`
  
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/check-for-build.jenkinsfile
Comment (gaiksaya) [inline jenkins/check-for-build.jenkinsfile:12]: We should remove 1.2.4 from this cron. The version is already out we do not need to keep rebuilding it

File: manifests/1.2.5/opensearch-1.2.5.yml
Comment (gaiksaya) [inline manifests/1.2.5/opensearch-1.2.5.yml:17]: I do not believe we add all components at first. Most of the components build might fail due to version incompatibility if the version is main is not incremented.

**Review by peternied — COMMENTED** [review]
We shouldn't merge this, after OpenSearch creates a 1.2.5 branch this manifest will be created automatically via this repos github actions

**Comment by abhinavGupta16** [comment]
> We shouldn't merge this, after OpenSearch creates a 1.2.5 branch this manifest will be created automatically via this repos github actions

I'll make this PR a draft

**Review by dblock — CHANGES_REQUESTED** [review]
This won't work. In order to produce a 1.2.5 successful build there needs to be version increments, then a -SNAPSHOT build with common-utils, then another with job-scheduler, etc. This is an incremental process that is semi-automated, but it runs reliable CI and ensures that we have a working 1.2.5 without ever being broken.

I would just close this without merging.

**Comment by abhinavGupta16** [comment]
Closing The PR since this process is automated

**Pull request closed without merging**
